hack/setenv.sh

```diff
@@ -0,0 +1,121 @@
+  if [ -n "$sa_namespace" ]; then
+    sa_namespace="$sa_namespace"
+  fi
+
+  read -p "What is the name of the SA? (default: $sa_name): " selected_name
+  if [ -n "$selected_name" ]; then
+    sa_name="$selected_name"
+  fi
+
+  if oc get namespace "$sa_namespace" &> /dev/null; then
+    echo "Namespace '$sa_namespace' already exists."
+  else
+    echo "Namespace '$sa_namespace' does not exist. Creating..."
+    oc create namespace "$sa_namespace"
+  fi
+
+  if oc get sa -n "$sa_namespace" $sa_name &> /dev/null; then
+    echo "ServiceAccount '$sa_name' already exists in '$sa_namespace'."
+  else
+    echo "ServiceAccount '$sa_name' does not exist in '$sa_namespace'. Creating..."
+    oc create sa "$sa_name" -n "$sa_namespace"
+  fi
+
+  oc adm policy add-cluster-role-to-user cluster-admin -z $sa_name -n $sa_namespace
+  echo "Added cluster-admin role to '$sa_name' in '$sa_namespace'."
+  token_secret=$(oc get secret -o name -n $sa_namespace | grep ${sa_name}-token)
+  token=$(oc get -n $sa_namespace ${token_secret} -o yaml | yq '.data.token' | sed 's/"//g' | base64 -d)
+  echo "export K8S_CLUSTER_TOKEN=$token" >> .env
+}
+
+function gitToken {
+  read -s -p "Enter GitHub access token: " value
+  echo ""
+  echo "export GITHUB_TOKEN=$value" >> .env
+}
+
+function argoCdNamespace {
+  default="orchestrator-gitops"
+  read -p "Enter ArgoCD installation namespace (default: $default): " value
+
+  if [ -z "$value" ]; then
+      argocd_namespace="$default"
+  else
+      argocd_namespace="$value"
+  fi
+  echo "export ARGOCD_NAMESPACE=$argocd_namespace" >> .env
+}
+
+function argoCdRoute {
+  argocd_instances=$(oc get argocd -n "$argocd_namespace" -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
+
+  if [ -z "$argocd_instances" ]; then
+      echo "No ArgoCD instances found in namespace $argocd_namespace."
+      exit 1
+  fi
+
+  echo "Select an ArgoCD instance:"
+  select instance in $argocd_instances; do
+      if [ -n "$instance" ]; then
+          selected_instance="$instance"
+          break
+      else
+          echo "Invalid selection. Please choose a valid option."
+      fi
+  done
+
+  argocd_route=$(oc get route -n $argocd_namespace -l app.kubernetes.io/managed-by=$selected_instance -ojsonpath='{.items[0].status.ingress[0].host}')
+  echo "Found Route at $argocd_route"
+  echo "export ARGOCD_URL=https://$argocd_route" >> .env
+  echo 
+}
+
+function argoCdCreds {
+  admin_password=$(oc get secret -n $argocd_namespace ${selected_instance}-cluster -ojsonpath='{.data.admin\.password}' | base64 -d)
+  echo "export ARGOCD_USERNAME=admin" >> .env
+  echo "export ARGOCD_PASSWORD=$admin_password" >> .env
+}
+
+function checkPrerequisite {
+  if ! command -v yq &> /dev/null; then
+    echo "yq is required for this script to run. Exiting."
+    exit 1
+  fi
+}
+
+checkPrerequisite
+workflowNamespace
+k8sUrl
+k8sToken
+gitToken
+argoCdNamespace
+argoCdRoute
+argoCdCreds
+
+cat .env
```

Comment:
why not `source` it instead of `cat` it? Maybe nice to save it in a file as well

Reply:
> I can source it :)
then .env file is already available at the end of the script execution

Yes I am stupid, I saw the `echo XXX >> .env` ...

Reply:
Too optimistic answer: I can't source it :doh:

After exporting the variables in the script, they are scoped to the context of the script itself. Once the script execution ends, the context reverts to the invoking shell, where the exported variables are not accessible. This is why sourcing the .env file at the end of the script hasn't any effect. To make the exported variables available in the invoking shell, you need to source the .env file directly in the shell where you're executing the script, rather than relying on the script to do it internally.

Reply:
would 
```
set -o allexport
source .env
set +o allexport
```
solve this?

Or maybe
```
export $(grep -v '^#' .env | xargs -d '\n')
``` 
source: https://stackoverflow.com/questions/19331497/set-environment-variables-from-file-of-key-value-pairs

Reply:
they don't work for me :/

Reply:
ok, let's forget about it then :)

Reply:
Can you remove ‘cat .env’ that was mainly added for debugging purposes?